app/controllers/duties_controller.rb

```diff
@@ -88,6 +92,17 @@ def grabable?(duty_id_params)
     end
   end
 
+  def drop_eligible?(drop_duty_ids)
+    duties = duties_sorted_by_start_time(drop_duty_ids)
+    first_timeslot = duties[0].timeslot_id
+    first_timerange = Timeslot.find(first_timeslot).time_range_id
+    first_start_time = TimeRange.find(first_timerange).start_time
+    date_limit = duties[0].date
+    time_limit = first_start_time - 7200
+    Time.zone.today <= date_limit &&
+      Time.zone.now.strftime("%H:%M:%S") < time_limit.strftime("%H:%M:%S")
```

Comment:
I’m sure that you can compare the time directly without the need to render to string first?

Reply:
I can do direct, hour-only comparison, but it would raise a small problem for opening hour duties as it would deny drops 30 minutes too early.

For an accurate comparison, the rendering to string method would be better because it compares down to seconds. It seems that it's the shortest way to extract the whole time from `start_time` (due to the nature of the `Time` class inserting dummy date to it). Other method is to directly comparing hours, then minutes, then seconds, but I think that would result in too many lines.

Reply:
Btw, I found something wrong in that code. Would be fixed in the next commit.

Reply:
Er, `<` doesn't work for our purposes. It also compares the dates, and `first_start_time` (subsequently, `time_limit`) contains a default date of `1 January 2000` for every timeslots so it would always return false. I guess the defaulting is a `Time` class behaviour.

Reply:
Can you combine the date from the `Duty`?